File: Julia/Arima/sarimaAirline.jl
# https://github.com/LAMPSPUC/StateSpaceModels.jl/blob/master/docs/src/examples.md

using CSV, DataFrames, StateSpaceModels, Plots

# Getting the data 
air_passengers = CSV.File("DataSet/Arima/airPassengers.csv") |> DataFrame

# Converting data for the plotting
log_air_passengers = log.(air_passengers.passengers)

# Plotting the data 
plot(log_air_passengers, title="Original Series", lindewidth=3)

# Applying SARIMA (Seasonal + ARIMA) to our data
model = SARIMA(log_air_passengers; order = (0, 1, 1), seasonal_order = (0, 1, 1, 12))
fit!(model)
print_results(model)

# Getting our forecast on our SARIMA Model
forec = forecast(model, 24)

# Plotting the forecasts results
plot(model, forec; legend = :topleft, title="SARIMA Prediction", label=["Original Series" "Prediction"], lindewidth=3)

File: Python/Arima/sarimaAirline.py
# https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.sarimax.SARIMAX.html
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt

# Getting the data
air_passengers = pd.read_csv("../../Dataset/Arima/airPassengers.csv")

# Converting data for plotting
log_air_passengers = np.log(air_passengers['passengers'])

# Plotting the data
plt.plot(log_air_passengers, label='Original Series', linewidth=1)
plt.title('Original Series')
plt.show()

# Applying SARIMA (Seasonal + ARIMA) to our data
model = sm.tsa.SARIMAX(log_air_passengers, order=(0, 1, 1), seasonal_order=(0, 1, 1, 12))
results = model.fit()

# Print estimates
print(results.summary())

# Getting our forecast on our SARIMA Model
forecast_values = results.get_forecast(steps=24)

# Plotting the forecast results
plt.figure(figsize=(10, 6))
plt.plot(log_air_passengers, label='Original Series', linewidth=1)
plt.plot(forecast_values.predicted_mean, label='Prediction', linewidth=1)
plt.title('SARIMA Prediction')
plt.legend(loc='upper left')
plt.show()
